pkgs/io_file/lib/src/vm_windows_file_system.dart

```diff
@@ -62,6 +67,81 @@ Exception _getError(int errorCode, String message, String path) {
   }
 }
 
+/// File system entity data available on Windows.
+final class WindowsMetadata extends Metadata {
+  // TODO(brianquinlan): Reoganize fields when the POSIX `metadata` is
+  // available.
+  // TODO(brianquinlan): Document the public fields.
+
+  final bool isReadOnly;
+  final bool isHidden;
+  final bool isSystem;
+  final bool isArchive;
+  final bool isTemporary;
+  final bool isOffline;
+  final bool isContentNotIndexed;
+
+  final int creationTime100Nanos;
+  final int lastAccessTime100Nanos;
+  final int lastWriteTime100Nanos;
+
+  DateTime get creation => _fileTimeToDateTime(creationTime100Nanos);
+  DateTime get access => _fileTimeToDateTime(lastAccessTime100Nanos);
+  DateTime get modification => _fileTimeToDateTime(lastWriteTime100Nanos);
+
+  /// TODO(bquinlan): Document this constructor.
+  WindowsMetadata({
+    super.isDirectory = false,
+    super.isFile = false,
+    super.isLink = false,
+
+    super.size = 0,
+
+    this.isReadOnly = false,
+    this.isHidden = false,
+    this.isSystem = false,
+    this.isArchive = false,
+    this.isTemporary = false,
+    this.isOffline = false,
+    this.isContentNotIndexed = false,
+
+    this.creationTime100Nanos = 0,
+    this.lastAccessTime100Nanos = 0,
+    this.lastWriteTime100Nanos = 0,
+  });
+
+  @override
+  bool operator ==(Object other) =>
+      other is WindowsMetadata &&
+      super == other &&
+      isReadOnly == other.isReadOnly &&
+      isHidden == other.isHidden &&
+      isSystem == other.isSystem &&
+      isArchive == other.isArchive &&
+      isTemporary == other.isTemporary &&
+      isOffline == other.isOffline &&
+      isContentNotIndexed == other.isContentNotIndexed &&
+      creationTime100Nanos == other.creationTime100Nanos &&
+      lastAccessTime100Nanos == other.lastAccessTime100Nanos &&
+      lastWriteTime100Nanos == other.lastWriteTime100Nanos;
+
+  @override
+  int get hashCode =>
+      (
+        super.hashCode,
+        isReadOnly,
+        isHidden,
+        isSystem,
+        isArchive,
+        isTemporary,
+        isOffline,
+        isContentNotIndexed,
+        creationTime100Nanos,
+        lastAccessTime100Nanos,
+        lastWriteTime100Nanos,
+      ).hashCode;
+}
+
 /// A [FileSystem] implementation for Windows systems.
 base class WindowsFileSystem extends FileSystem {
```

Comment:
Consider calling `GetLastError` in the constructor to avoid repeating it in each method.

```suggestion
base class WindowsFileSystem extends FileSystem {
  WindowsFileSystem() {
    // Calling `GetLastError` for the first time causes the `GetLastError`
    // symbol to be loaded, which resets `GetLastError`. So make a harmless
    // call before the value is needed.
    win32.GetLastError();
  }
```